src/v/kafka/server/handlers/topics/topic_utils.cc

```diff
@@ -65,8 +65,9 @@ ss::future<> wait_for_topics(
     return ss::do_with(
       std::move(results),
       [&md_cache, &api, timeout](std::vector<cluster::topic_result>& results) {
-          return ss::parallel_for_each(
+          return ss::max_concurrent_for_each(
                    results,
+                   5,
```

Comment:
why 5, maybe a bit bigger number can be used, like 32 ?

Reply:
couple of reasons I did this.

* actual concurrency is controlled by num partitions, #topics * std::min(num_partitions, 4096), 32 * 4096 seemed like a lot of tasks in one shot.
* unlikely that users create multiple topics at once with this API, so in most cases size(topics) == 1 or some low number, so it doesn't really matter.